shm/Player.hpp

```diff
@@ -13,6 +13,7 @@ class Player {
     size_t getAvaliableSpace() const { return avaliableSpace_; };
     size_t getSpeed() const { return ship_->getSpeed(); };
     Cargo* getCargo(size_t index) const { return ship_->getCargo(index); };
+    std::vector<std::shared_ptr<Cargo>> getAllCargos() const { return ship_->getAllCargos(); }
```

Comment:
```suggestion
    std::vector<std::shared_ptr<Cargo>>& getAllCargos() const { return ship_->getAllCargos(); }
```

So it's not copied every time

Reply:
Yes, yes - thanks!

Reply:
But after add reference i got this error "initial value of reference to non-const must be an lvalue" and I don't have idea why.

Reply:
Possibly method which uses this function is not declared as const

Reply:
When I write it like this: `const std::vector<std::shared_ptr<Cargo>>&` both in Ship and Player there is no error, but it's correct?

Reply:
I would say it's ok since it's just a getter

Reply:
I guess that `ship_->getAllCargos(); ` should have the same return type as`getAllCargos() ` in Player class  so both should be `std::vector<std::shared_ptr<Cargo>>&`